gmso/formats/mol2.py

```diff
@@ -49,52 +50,76 @@ def from_mol2(filename, site_type="atom"):
     # Initialize topology
     topology = Topology(name=mol2path.stem)
     # save the name from the filename
-    with open(mol2path, "r") as f:
-        line = f.readline()
-        with open(filename, "r") as f:
-            fcontents = f.readlines()
-        sections = dict()
-        key = "meta"
-        for line in fcontents:
-            if "@<TRIPOS>" in line:
-                section_key = line.strip("\n")
-                sections[section_key] = list()
-            else:
-                sections[section_key].append(line)
-
-        supported_rti = {
-            "@<TRIPOS>ATOM": _parse_atom,
-            "@<TRIPOS>BOND": _parse_bond,
-            "@<TRIPOS>CRYSIN": _parse_box,
-            "@<TRIPOS>FF_PBC": _parse_box,
-        }
-        for section in sections:
-            if section not in supported_rti:
-                warnings.warn(
-                    f"The record type indicator {section} is not supported."
-                )
-            else:
-                supported_rti[section](
-                    topology, sections[section], site_type="atom"
-                )
+    with open(filename, "r") as f:
+        fcontents = f.readlines()
+    sections = dict()
+    for line in fcontents:
+        if "@<TRIPOS>" in line:
+            section_key = line.strip("\n")
+            sections[section_key] = list()
+        else:
+            sections[section_key].append(line)
+
+    _parse_site = _parse_lj if site_type == "lj" else _parse_atom
+    supported_rti = {
+        "@<TRIPOS>ATOM": _parse_site,
+        "@<TRIPOS>BOND": _parse_bond,
+        "@<TRIPOS>CRYSIN": _parse_box,
+        "@<TRIPOS>FF_PBC": _parse_box,
+    }
+    for section in sections:
+        if section not in supported_rti:
+            warnings.warn(
+                f"The record type indicator {section} is not supported."
+            )
+        else:
+            supported_rti[section](
+                topology, sections[section], site_type="atom"
```

Comment:
Just need to remove site_type argument now to this function call. LGTM after this.